ldp.gemspec

```diff
@@ -10,30 +10,37 @@ Gem::Specification.new do |spec|
   spec.email         = ["chris@cbeer.info"]
   spec.description   = %q{Linked Data Platform client library}
   spec.summary       = spec.description
-  spec.homepage      = "https://github.com/projecthydra/ldp"
+  spec.homepage      = "https://github.com/samvera/ldp"
   spec.license       = "APACHE2"
-  spec.required_ruby_version = '>= 2.0'
+  spec.required_ruby_version = '>= 2.4'
 
   spec.files         = `git ls-files`.split($/)
   spec.executables   = spec.files.grep(%r{^bin/}) { |f| File.basename(f) }
   spec.test_files    = spec.files.grep(%r{^spec/})
   spec.require_paths = ["lib"]
 
+  spec.add_dependency "deprecation"
   spec.add_dependency "faraday", '>= 1'
-  spec.add_dependency "rdf",            ">= 1.1"
-  spec.add_dependency "rdf-turtle"
-  spec.add_dependency "rdf-vocab",      ">= 0.8"
-  spec.add_dependency "rdf-isomorphic"
-  spec.add_dependency "json-ld"
   spec.add_dependency "http_logger"
-  spec.add_dependency "deprecation"
+  spec.add_dependency "json-ld", "3.2.1"
+  spec.add_dependency "rdf", "3.2.1"
```

Comment:
You are intentionally locking to _exact_ versions of json-ld and rdf gems?  That is unusual, so just confirming it's intentional. It means that any app with `ldp` in it's dependency tree can _only_ use exactly 3.2.1 of both json-ld and `rdf`  gems; 

if something else in the dependency tree instead were to instead demand say, `3.2.2` exactly (which is out!), it would be an unresolvable conflict -- so if any other dependencies decide to start locking to specific versions like this, it's likely to create unresolvable dependency conflicts.

Reply:
Loosening these dependencies with https://github.com/samvera/ldp/compare/allow-ruby-3.0-cjc...allow-ruby-3.0-jrgriffiniii?expand=1 would require that Ruby 2.7 and Rails 5.2 no longer be supported: https://app.circleci.com/pipelines/github/samvera/ldp/422/workflows/ad63f37a-08c9-41bc-ac1d-1c3f8ca079a1

Reply:
I restarted the failing test and it passed so I think it is safe to loosen those restrictions.

Reply:
If needed you could also *test* under (eg) ruby 2.7 and Rails 5.2 with an earlier version than is actually allowed by the spec. You could have a gemspec that allows versions that don't work with ruby 2.7/rails 5.2, but still support them so long as the gemspec *also* allows versions that *do* work with that combo -- although it requires people using that old combo to know to restrict to versions that work for it in their Gemfile. But that's what happens sometimes using EOL software (as rails 5.2 is). 

That's what I'd advise if it were necessary, but sounds like it may not be necessary! Thanks!

Reply:
Except it looks like you've already merged with the specific dependency? 

Up to you, but my feedback is that's not a good idea, gems requiring single specific versions of intermediate dependencies makes dependency management extremely hard for apps. And will prevent, eg, an app from using a subsequent release with a security patch as well.